src/twisted/conch/ssh/keys.py

```diff
@@ -1241,8 +1402,20 @@ def verify(self, signature, data):
                 k = k.public_key()
             verifier = k.verifier(
                 signature, hashes.SHA1())
-        else:
-            raise BadKeyError("unknown key type %s" % (self.type(),))
+
+        elif keyType == 'EC':
+            k = self._keyObject
+            if not self.isPublic():
+                k=k.public_key()
```

Comment:
This comment still hasn't been addressed; codecov shows it as uncovered: https://codecov.io/gh/twisted/twisted/compare/620b84494a0d59309d27c1228bc3886771051a32...b78e520b5f7e196b4a45ba649042918744b56d48#7372632F747769737465642F636F6E63682F7373682F6B6579732E7079-1404